docs/content/manual/manual.md

```diff
@@ -0,0 +1,5424 @@
+
+The `search` key in the metadata, if present, should have a
+string or array value (array of strings); this is the search
+path to be prefixed to the top-level search path.
+
+### Importing data
+
+The directive
+
+    import RelativePathString as $NAME [<metadata>];
+
+imports a JSON file found at the given path relative to a directory in a search path.
+A `.json` suffix will be added to the relative path string.
+The file's data will be available as `$NAME::NAME`.
+
+The optional metadata is considered the same way as [module imports](#importing-modules).
+
+### Providing module metadata
+
+The directive
+
+    module <metadata>;
+
+may be put at the beginning of a module file.
+It is entirely optional and serves only the purpose of providing
+metadata that can be read with the [`modulemeta`](#modulemeta) builtin.
+
+The metadata must be a constant jq expression.  It should be
+an object with keys like `homepage`.  At this time jq doesn't
+use this metadata, but it is made available to users via the
+`modulemeta` builtin.
+
+### Search paths
+
+Modules imported by a program are searched for in a default search path (see below).
+The `import` and `include` directives allow the importer to alter this path.
+
+Paths in the search path are subject to various substitutions:
+
+* For paths starting with `~/`, the user's home directory is substituted for `~`.
+* For paths starting with `$ORIGIN/`,
+  the directory where the jq executable is located is substituted for `$ORIGIN`.
+* For paths starting with `./` or paths that are `.`,
+  the path of the including file is substituted for `.`.
+  For top-level programs given on the command-line, the current directory is used.
+
+Import directives can optionally specify a search path to which
+the default is appended.
+
+The default search path is the search path given to the `-L` command-line option,
+else `["~/.jq", "$ORIGIN/../lib/jq", "$ORIGIN/../lib"]`.
+
+Null and empty string path elements terminate search path processing.
+
+A dependency with relative path `foo/bar` would be searched for in
+`foo/bar.jq` and `foo/bar/bar.jq` in the given search path.
+This is intended to allow modules to be placed in a directory along with, for example,
+version control files, README files, and so on, but also to allow for single-file modules.
+
+Consecutive components with the same name are
+not allowed to avoid ambiguities (e.g., `foo/foo`).
+
+For example, with `-L$HOME/.jq` a module `foo` can be found in
+`$HOME/.jq/foo.jq` and `$HOME/.jq/foo/foo.jq`.
+
+If `.jq` exists in the user's home directory, and is a file (not a directory),
+it is automatically sourced into the main program.
+
+
+
+# Builtin functions
+
+This section documents all named filter functions that are
+available by default in any jq program.
+
+## Basic functions
+
+### `empty`
+
+`empty` returns no results. None at all. Not even `null`.
+
+It's useful on occasion. You'll know if you need it :)
```

Comment:
Maybe we could mention backtracking and use in recursive functions here? or refer to the section about functions and recursion?

Reply:
I personally do not see a special link between `empty` and backtracking/recursive functions. `empty` can be useful in so many ways. I'm actually not sure whether `empty` or `.` is my favourite filter. ;)
I would leave it as it is for now, given that this is what the original docs say about `empty`.